torch/ao/sparsity/experimental/data_sparsifier/lightning/callbacks/data_sparsity.py

```diff
@@ -0,0 +1,59 @@
+from copy import deepcopy
+import torch
+from dataclasses import dataclass, field
+from typing import Any, Optional, Dict
+import pytorch_lightning as pl
+
+from ._data_sparstity_utils import _attach_model_to_data_sparsifier, _log_sparsified_level
+
+
+@dataclass
+class _DataSparsity(pl.callbacks.Callback):
+    data_sparsifier_type: Any
+    data_sparsifier_args: Dict  # this the arguments for sparsifier_type (except data_list)
+    data_sparsifier: Any = field(init=False, default=None)
+    sparsified: Optional[torch.nn.Module] = field(init=False, default=None)
+
+    def __post_init__(self):
+        pass
+
+
+@dataclass
+class PostTrainingDataSparsity(_DataSparsity):
+    """Lightning callback that enables post-training sparsity.
+
+    This callback aims to sparsify the model inside lightning module after training.
+    **Note that the model is copied and then sparsified, so the existing model is not modified**
+
+    The sparsified model can be used for comparison and can be accessed using
+        <callback_obj>.sparsified
+
+    Args:
+        data_sparsifier_type (some implemented class of BaseDataSparsifier)
+            The data sparsifier object of this type is created when the
+            training starts.
+            Note: Objects should not be passed in here as they are created
+            once the training completes.
+
+        data_sparsifier_args (Dict)
```

Comment:
wondering if you would like to adopt the `with_args` design: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/observer.py#L1460

Reply:
in this case `data_sparsifier_type` should be named as `data_sparsifier_constructor` I think